types/aws-lambda/trigger/cognito-user-pool-trigger/index.d.ts

```diff
@@ -0,0 +1,49 @@
+import { Handler } from '../../handler';
+import { CreateAuthChallengeTriggerEvent } from './create-auth-challenge';
+import { CustomMessageTriggerEvent } from './custom-message';
+import { DefineAuthChallengeTriggerEvent } from './define-auth-challenge';
+import { PostAuthenticationTriggerEvent } from './post-authentication';
+import { PostConfirmationTriggerEvent } from './post-confirmation';
+import { PreAuthenticationTriggerEvent } from './pre-authentication';
+import { PreSignUpTriggerEvent } from './pre-signup';
+import { PreTokenGenerationTriggerEvent } from './pre-token-generation';
+import { UserMigrationTriggerEvent } from './user-migration';
+import { VerifyAuthChallengeResponseTriggerEvent } from './verify-auth-challenge-response';
+
+/**
+ * Cognito User Pool event
+ * @deprecated Please use specific event types instead
+ * @see http://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools-working-with-aws-lambda-triggers.html
+ */
+export type CognitoUserPoolTriggerEvent =
```

Comment:
This is in intersection type because the current implementation is equivalent of this intersection type. 
Ideally this should be a union type. but, that'd break the backward-compatibility.